Adds address details page

s-ben opened the pull request:
This PR creates a 'Address details' page under /Advanced ( Closes #779 , Closes #710 ). Previously, technical details about addresses were found in the glossary definition for Address and a couple FAQ questions. An advanced page seemed like a better place to aggregate this information. Have also added the table @zubairzia0 added in Issue https://github.com/decred/dcrdocs/issues/710 (@jholdstock maybe this finally closes that issue?). The glossary definition of address and relevant FAQ questions have also been updated to be more concise and point to the new address page.

jholdstock commented on docs/faq/wallets-and-seeds.md:
```diff
@@ -70,27 +70,7 @@ A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with i
 
 #### 8. What are the different types of addresses?
 
-A Decred address[^14995] is actually just a representation of a public key (which itself could be a script hash) along with a 2-byte prefix which identifies the network and type and a checksum suffix in order to detect improperly entered addresses.
```
> a checksum suffix in order to detect improperly entered addresses.

The new page loses this detail. Can we include it please? Maybe including how many bytes the checksum consists of.

s-ben replied:
Good catch. Was meaning to include that, but got lost in my wordsmithing:)

Have added detail to first para:

> Each address has a 2-byte prefix than can be used to identify its type, and a checksum suffix to detect improperly entered addresses. The first byte of the prefix identifies the network...

s-ben replied:
I can also add the number of bytes the checksum uses if someone can provide. I assume it's 4 like Bitcoin addresses, but don't want to assume and can't find this in our existing docs.

davecgh replied:
2-byte prefix, fingerprint, 4-byte checksum.  The checksum is 4 bytes of `blake256(blake256(prefix||checksum))`.

jholdstock commented on docs/faq/wallets-and-seeds.md:
```diff
@@ -70,27 +70,7 @@ A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with i
 
 #### 8. What are the different types of addresses?
 
-A Decred address[^14995] is actually just a representation of a public key (which itself could be a script hash) along with a 2-byte prefix which identifies the network and type and a checksum suffix in order to detect improperly entered addresses.
```
As we are dropping this line, we no longer need the reference [^14995] at the bottom of this file. We can delete that line as well

jholdstock commented on docs/faq/wallets-and-seeds.md:
```diff
@@ -70,27 +70,7 @@ A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with i
 
 #### 8. What are the different types of addresses?
```
Seeing as we now have both the glossary entry and the whole page dedicated to addresses, do you think we could just drop this FAQ entry altogether? To me it feels like we have enough content on this subject and we dont need the FAQ as well

s-ben replied:
If this was only in a page under /Advanced, I'd say keep it, as it's not easily discoverable. But with the address definition in the glossary (which links to this page), I think we should remove this (have removed in last push, but can add back if someone strongly disagrees).

jholdstock commented on docs/faq/wallets-and-seeds.md:
```diff
@@ -70,27 +70,7 @@ A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with i
-| Pay-to-Pubkey-Hash (secp256k1)  	|       P2PKH      	|         Ds        	|    Ts   	|   Ss   	|    2                  	|
-| Pay-to-Script-Hash 	|       P2SH       	|         Dc        	|    Tc   	|   Sc   	|    2                  	|
-
----
```
If we decide to keep this FAQ entry, this page break should be re-instated. If the FAQ is removed, the page break can go too.

s-ben commented on docs/advanced/address-details.md:
````diff
@@ -0,0 +1,31 @@
+
+Each address has a 2-byte prefix than can be used to identify its type. The first byte of the prefix identifies the network. Mainnet addresses start with `D`, testnet addresses start with `T`, simnet addresses start with `S`, and regnet addresses start with `R`. The second byte identifies the address type. Pay-to-PubkeyHash (P2PKH) addresses, for example, contsin a lowercase `s` in the second byt ...
+
+```DsYszZ5n79HJzyTVGAdXeDb1b3Z5N2Tnsij```
````
Changed address to example provided above,
``````DsExampleAddr1For2Demo3PurposesOnly```

A testnet might look better, but couldn't find one easily. If someone posts one here can use that instead.

jholdstock replied:
Here is a valid testnet address `TsSpRM32SQLkWQk6WPkB1p8PcrCujdSkf5C`

s-ben commented:
@jholdstock have addressed all comments. I do think a more detailed reference like the [bitcoin address page](https://en.bitcoinwiki.org/wiki/Technical_background_of_version_1_Bitcoin_addresses) you point out would be good. I know @davecgh likes them diagrams:) Would be happy to at least update the Bitcoin diagram for Decred-specific deviations.
![image](https://user-images.githubusercontent.com/1634777/51638521-63c1dd80-1f13-11e9-854b-b459d88db24c.png)
